autodispose/src/main/java/com/uber/autodispose/KotlinExtensions.kt

```diff
@@ -253,3 +254,48 @@ inline fun <T> ParallelFlowable<T>.autoDisposable(provider: ScopeProvider): Para
 @CheckReturnValue
 inline fun <T> ParallelFlowable<T>.autoDispose(provider: ScopeProvider): ParallelFlowableSubscribeProxy<T> =
     this.`as`(AutoDispose.autoDisposable(provider))
+
+/** Executes a [body] with under an [AutoDisposeContext] backed by the given [scope]. */
+fun withScope(scope: ScopeProvider, body: AutoDisposeContext.() -> Unit) {
+  withScope(completableOf(scope), body)
+}
+
+/** Executes a [body] with under an [AutoDisposeContext] backed by the given [completableScope]. */
+fun withScope(completableScope: Completable, body: AutoDisposeContext.() -> Unit) {
+  val context = RealAutoDisposeContext(completableScope)
+  context.body()
+}
+
+/**
+ * A context intended for use as `AutoDisposeCoroutineScope.() -> Unit` function body parameters
```

Comment:
Given that this is in the `autodispose` module and that you can use it with a normal `ScopeProvider` as well, we can probably skip the reference to coroutines.

Reply:
That's a stale doc from the original PR, will fix

Reply:
AutoDisposeContext